File: ch2_labs/lab6_C++/TextLibrary/TextRow.cpp
#include "TextRow.h"
#include <algorithm>

TextRow::TextRow(const std::string& text) : content(text) {}

std::string TextRow::GetContent() const {
    return content;
}

void TextRow::ReplaceChar(char oldChar, char newChar) {
    for (char& ch : content)
        if (ch == oldChar) ch = newChar;
}

int TextRow::CountChar(char c) const {
    return std::count(content.begin(), content.end(), c);
}

int TextRow::Length() const {
    return content.size();
}

File: ch2_labs/lab6_C#/TextLibrary/TextRow.cs
public class TextRow
{
    private string content;

    public TextRow(string text)
    {
        content = text;
    }

    public string Content => content;

    public void ReplaceChar(char oldChar, char newChar)
    {
        content = content.Replace(oldChar, newChar);
    }

    public int CountChar(char c)
    {
        return content.Count(ch => ch == c);
    }

    public int Length => content.Length;
}
